source/georef/path_finder.cpp

```diff
@@ -0,0 +1,517 @@
+        distances[starting_edge[target_e]] = crow_fly_duration(starting_edge.distances[target_e]);
+        predecessors[starting_edge[source_e]] = starting_edge[source_e];
+        predecessors[starting_edge[target_e]] = starting_edge[target_e];
+
+        if (starting_edge[target_e] != starting_edge[source_e]) {  // if we're on a useless edge we do not enhance
+            // small enchancement, if the projection is done on a node, we disable the crow fly
+            if (starting_edge.distances[source_e] < 0.01) {
+                predecessors[starting_edge[target_e]] = starting_edge[source_e];
+                distances[starting_edge[target_e]] = bt::pos_infin;
+            } else if (starting_edge.distances[target_e] < 0.01) {
+                predecessors[starting_edge[source_e]] = starting_edge[target_e];
+                distances[starting_edge[source_e]] = bt::pos_infin;
+            }
+        }
+    }
+
+    if (color.n != n) {
+        color = boost::two_bit_color_map<>(n);
+    }
+}
+
+std::pair<navitia::time_duration, ProjectionData::Direction> PathFinder::find_nearest_vertex(
+    const ProjectionData& target,
+    bool handle_on_node) const {
+    constexpr auto max = bt::pos_infin;
+    if (!target.found)
+        return {max, source_e};
+
+    if (distances[target[source_e]] == max)  // if one distance has not been reached, both have not been reached
+        return {max, source_e};
+
+    if (handle_on_node) {
+        // handle if the projection is done on a node
+        if (target.distances[source_e] < 0.01) {
+            return {distances[target[source_e]], source_e};
+        } else if (target.distances[target_e] < 0.01) {
+            return {distances[target[target_e]], target_e};
+        }
+    }
+
+    auto source_dist = distances[target[source_e]] + crow_fly_duration(target.distances[source_e]);
+    auto target_dist = distances[target[target_e]] + crow_fly_duration(target.distances[target_e]);
+
+    if (target_dist < source_dist)
+        return {target_dist, target_e};
+
+    return {source_dist, source_e};
+}
+
+Path PathFinder::get_path(type::idx_t idx) {
+    if (!computation_launch)
+        return {};
+    ProjectionData projection = this->geo_ref.projected_stop_points[idx][mode];
+
+    auto nearest_edge = find_nearest_vertex(projection);
+
+    return get_path(projection, nearest_edge);
+}
+
+void PathFinder::add_custom_projections_to_path(Path& p,
+                                                bool append_to_begin,
+                                                const ProjectionData& projection,
+                                                ProjectionData::Direction d) const {
+    auto item_to_update = [append_to_begin](Path& p) -> PathItem& {
+        return (append_to_begin ? p.path_items.front() : p.path_items.back());
+    };
+    auto add_in_path = [append_to_begin](Path& p, const PathItem& item) {
```

Comment:
I don't know if it would make things simpler by using `std::front_inserter` or `std::back_inserter` with something like: 
```cpp
auto path_inserter = append_to_begin? std::front_inserter(p.path_tiems) : std::back_inserter(p.path_tiems);
[...]
path_inserter = item;
```

In a similar way, you could have to update your items:
```cpp
auto item_to_update = append_to_begin ? 
                     [&](){ return p.path_items.front(); } : 
                     [&](){ return p.path_items.back(); } ;
[...]
item_to_update().way_id 
```

Those are just ideas to make things nicer, feel free to disregard.
#ModernizeMyCpp